src/integrations/MySQL2Integration.ts

```diff
@@ -79,33 +114,49 @@ class MySQL2Integration implements Integration {
                         [DBTags.DB_TYPE]: DBTypes.MYSQL,
                         [SpanTags.TOPOLOGY_VERTEX]: true,
                     });
-
-                    const sequence = query.call(this, sql, values, cb);
-
-                    const statement = sequence.sql;
-
-                    if (statement) {
-                        const statementType = statement.split(' ')[0].toUpperCase();
+                    if (queryString) {
+                        const statementType = queryString.split(' ')[0].toUpperCase();
                         span.addTags({
                             [DBTags.DB_STATEMENT_TYPE]: statementType,
-                            [DBTags.DB_STATEMENT]: config.maskRdbStatement ? undefined : statement,
-                            [SpanTags.OPERATION_TYPE]: SQLQueryOperationTypes[statementType] ?
-                                SQLQueryOperationTypes[statementType] : '',
+                            [DBTags.DB_STATEMENT]: config.maskRdbStatement ? undefined : queryString,
+                            [SpanTags.OPERATION_TYPE]: SQLQueryOperationTypes[statementType]
+                                ? SQLQueryOperationTypes[statementType]
+                                : '',
                         });
                     }
 
                     span._initialized();
 
-                    const originalCallback = sequence.onResult;
+                    // Wrap the query and then let it execute
 
-                    const wrappedCallback = (err: any, res: any) => {
+                    const wrappedCallback = (err: any, res: any, fields: any) => {
                         if (err) {
                             span.setErrorTag(err);
                         }
                         ThundraLogger.debug(`<MySQL2Integration> Closing MySQL span with name ${span.getOperationName()}`);
-                        span.closeWithCallback(me, originalCallback, [err, res]);
+                        span.closeWithCallback(this, originalCallback, [err, res, fields]);
                     };
 
+                    if (sql.onResult) {
+                        sql.onResult = wrappedCallback;
+                    } else {
+                        callback = wrappedCallback;
+                    }
+                    if (overrideInnerCallback) {
+                        sql._callback = wrappedCallback;
+                    }
+
+                    return query.call(this, sql, params, callback);
+
+                    /*
```

Comment:
Should clean up  comment-out section on code base

Reply:
Removed